util/test/checkChplInstall

```diff
@@ -23,213 +24,146 @@
-elif [ ! -x $CHPL_BIN ] ; then
-    log_error "Found ${bin_name} at ${CHPL_BIN} but it is not executable."
+elif [ ! -x ${CHPL_BIN} ] ; then
+    err "Found ${BIN_NAME} at ${CHPL_BIN} but it is not executable."
     exit 1
 else
-    log_debug "Found ${bin_name} at: ${CHPL_BIN}"
+    echo "Found executable ${BIN_NAME} in ${CHPL_BIN}."
 fi
 
 # Verify CHPL_HOME is correctly set.
 if [ -z "${CHPL_HOME+x}" ] ; then
-    log_error "CHPL_HOME is not set in environment."
+    err "CHPL_HOME is not set in environment."
     exit 1
-elif [ ! -d $CHPL_HOME ] ; then
-    log_error "CHPL_HOME is not a directory: ${CHPL_HOME}"
+elif [ ! -d ${CHPL_HOME} ] ; then
+    err "CHPL_HOME is not a directory: ${CHPL_HOME}"
     exit 1
 else
-    log_debug "CHPL_HOME is set to: ${CHPL_HOME}"
+    echo "Found \$CHPL_HOME directory: ${CHPL_HOME}"
+fi
+
+# Create ~/.chpl directory, if it does not already exist
+CHPL_DIR=${CHPL_CHECK_INSTALL_DIR:-${HOME}/.chpl}
+if [ ! -d ${CHPL_DIR} ] ; then
+    echo "${CHPL_DIR} does not exist. Creating it."
+    mkdir -p ${CHPL_DIR} || { echo "Failed to create ${CHPL_DIR}." && exit 10 ; }
 fi
 
-# Find examples test directory.
-if [ -d $CHPL_HOME/test/release/examples ] ; then
+# Location of test job
+if [ -d ${CHPL_HOME}/test/release/examples ] ; then
     # Install from source repo.
-    TEST_DIR=$CHPL_HOME/test/release/examples
-elif [ -d $CHPL_HOME/examples ] ; then
+    TEST_DIR=${CHPL_HOME}/test/release/examples
+elif [ -d ${CHPL_HOME}/examples ] ; then
     # Install from tarball.
-    TEST_DIR=$CHPL_HOME/examples
+    TEST_DIR=${CHPL_HOME}/examples
 else
-    log_error "Could not find test cases in CHPL_HOME: ${CHPL_HOME}."
+    err "Could not find test cases in CHPL_HOME: ${CHPL_HOME}."
     exit 1
 fi
 
-# For chpldoc, only test the primers/chpldoc.doc.chpl test.
-if [ "${bin_name}" = "chpldoc" ] ; then
-    test_name="chpldoc"
-    TESTS=$TEST_DIR/primers/chpldoc.doc.*
+TMP_TEST_DIR="$(mktemp -d ${CHPL_DIR}/chapel-test-XXXXX)"
+echo "Temporary test job directory: ${TMP_TEST_DIR}"
 
-# List all the files in the examples dir. Do not list any subdirectories,
-# though.
-else
-    test_name="hello world"
-    TESTS=$(find $TEST_DIR -maxdepth 1 -mindepth 1 -type f)
-fi
+TEST_JOB=${TEST_JOB_BASENAME}
 
-# Copy these tests to a temp file.
+# Collect comm protocol environment variables (lowercase to avoid conflicts)
+chpl_comm="$($CHPL_HOME/util/chplenv/chpl_comm.py)"
+chpl_launcher="$($CHPL_HOME/util/chplenv/chpl_launcher.py)"
 
-chpl_dir=${CHPL_CHECK_INSTALL_DIR:-$HOME/.chpl}
-if [ ! -d $chpl_dir ] ; then
-    log_debug "${chpl_dir} does not exist. Creating it."
-    mkdir -p $chpl_dir || { log_error "Failed to create ${chpl_dir}." && exit 10 ; }
-fi
+# Log file for compile output
+TEST_COMP_OUT=${TEST_JOB}.comp.out
 
-TMP_TEST_DIR=$(mktemp -d ${chpl_dir}/chapel-test-XXXXX)
-log_debug "TMP_TEST_DIR: ${TMP_TEST_DIR}"
+# Necessary to suppress warnings when WARNINGS=1 (e.g. nightly build settings)
+COMP_FLAGS="--cc-warnings"
```

Comment:
Hey @ben-albrecht — Over in [this comment](https://github.com/chapel-lang/chapel/issues/15489#issuecomment-646665155) a user is running into some compilation problems due to our throwing `-Werror` (and friends) during the make check step.  Those issues are worth digging into (and I am), but I was surprised that we throw `-Werror`-style flags during that step, as we don't typically throw them for a Chapel compile, only in our own internal testing to try and keep our code as clean as possible.  To me, the comment here seems incorrect, as my understanding of `--cc-warnings` is that it says "add these stricter flags" not "suppress warnings".  Do you see any issue with my removing it?

Reply:
@bradcray - I agree that the comment is incorrect, and I see no issue with removing the `--cc-warnings`.

I definitely don't trust this code too much since it was written during my first week on the team 😄.

Reply:
Oh wow!  :O

Thanks!